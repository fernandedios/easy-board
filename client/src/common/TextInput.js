import React, { Component } from 'react';

class TextInput extends Component {
    renderIcon() {
        if (this.props.message !== '') {
            return <span className="icon is-small is-right"><i className="fas fa-exclamation-triangle"></i></span>;
        }
    }

    render() {
        const { name, label, message, placeholder, ...others } = this.props;

        return (
            <div className="field">
                <label className="label">{label}</label>
                    <div className="control has-icons-right">
                        <input {...others} id={name} name={name} className="input" placeholder={placeholder} />
                        {this.renderIcon()}
                    </div>
                <p className="help is-danger">{message}</p>
            </div>
        );
    }
}

TextInput.defaultProps = {
    message: ''
};

export default TextInput;
