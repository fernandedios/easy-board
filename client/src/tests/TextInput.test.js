import React from 'react';
import { shallow, mount } from 'enzyme';
import ReactTestRenderer from 'react-test-renderer';

import TextInput from '../common/TextInput';

describe('<TextInput />', () => {
    it('should be defined', () => {
        expect(TextInput).toBeDefined();
    });

    it('should have the correct classNames', () => {
        const output = shallow(<TextInput name="test" />);
        expect(output.find('input').hasClass('input')).toEqual(true);
    });

    it('should render input field correctly', () => {
        const output = mount(<TextInput type="text" name="test" placeholder="test placeholder" />);
        expect(output.find('input').prop('placeholder')).toEqual('test placeholder');
        expect(output.find('input').prop('type')).toEqual('text');
    });

    it('should render correctly', () => {
        const output = ReactTestRenderer.create(<TextInput name="test" />).toJSON();
        expect(output).toMatchSnapshot();
    });
});
