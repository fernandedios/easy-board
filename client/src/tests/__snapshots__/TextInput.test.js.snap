// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TextInput /> should render correctly 1`] = `
<div
  className="field"
>
  <label
    className="label"
  />
  <div
    className="control has-icons-right"
  >
    <input
      className="input"
      name="test"
      placeholder={undefined}
    />
  </div>
  <p
    className="help is-danger"
  >
    
  </p>
</div>
`;
